server/cast/text.go

```diff
@@ -68,4 +68,11 @@ func textImplicit() {
 			return handleStringCast(val.(string), targetType)
 		},
 	})
+	framework.MustAddImplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.Text,
+		ToType:   pgtypes.Regclass,
+		Function: func(ctx *sql.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			return targetType.IoInput(ctx, val.(string))
+		},
+	})
```

Comment:
Let's move this right above `VarChar` since they're always in alphabetical order, and will make it easier to search for it if we ever need to modify it.